### ISSUE TITLE
On the tablets UIkit thinks that device resolution is small.

### ISSUE THREAD

**Issue by xonixmx** [uikit/uikit#3973]
On the tablets UIkit thinks that device resolution is small. What I mean. When I use uk-child-width-1-2@s uk-child-width-1-5@m 
the style ...@s will works even the tablet in landscape mode and resolution FullHD and style @m will not works. UIkit thinks that tablet always has the small screen resolutions. I guess something wrong with @media settings in your css.

**Comment by andy-blum** [comment]
This is really hard to debug without providing a link to the issue.

**Comment by zzseba78** [comment]
What tablet model are you using? what is the screen size points?
Check this article if it gives you some light https://www.electricmayhemsolutions.net/screen-resolution-vs-viewport-responsive-web-design/

**Comment by xonixmx** [comment]
Yes, I have <meta name="viewport" content="width=device-width, initial-scale=1"> in my html code. I've Huawei tablet model MediaPad M3 with 2560x1600 screen resolution and if I remember correct that has 359 ppi. Anyway, when I try to use @s, @m, etc. it always behave as in a small screen resulution. Also as I can see the same situation on movile phone in landscape mode - it does't care that screen size isn't small. I guess it's thinks - oh, it's mobile divece, it has small screens as I can see, it's not a PC's display and so on. I just try to use Grid and Card objects. They behaves as I said and I have to use @media in CSS with digits parameters for screens for that situation, but I wanna use @s, @m, etc. for this.

**Comment by xonixmx** [comment]
The problem in that -- params as @s @m @l, etc. do not take into account the device pixel ratio. It's only correct works if you have standard PC screen with devicePixelRatio = 1. I have smartphone with devicePixelRatio = 2 and tablet with devicePixelRatio = 3 and I have two another situations by this reason.

**Comment by andy-blum** [comment]
@xonixmx Are you including `<meta name="viewport" content="width=device-width, initial-scale=1">` in your HTML <head>?

**Comment by andy-blum** [comment]
Just to double check you definitely have  `content="width=device-width, initial-scale=1"` in that viewport metatag, right?

That's super important to make sure the browser doesn't render the site on a virtual viewport and then scale it to fit.

[Additional reading](https://developer.mozilla.org/en-US/docs/Mozilla/Mobile/Viewport_meta_tag)

**Comment by xonixmx** [comment]
I have 
`<meta name="viewport" content="width=device-width, initial-scale=1">`

**Comment by xonixmx** [comment]
Okay. Please open my website (it's under construction now). Here's a link [](http://en.div.st) . (the login and password: 123) Just scroll to the CV section and I have two qestion: on my PC I can see two cols per row but on tablet (landscape mode) only one. And please see the image stamps under CV text blocks -- on PC it's five on the one line, but on tablet just three and two.

**Comment by andy-blum** [comment]
That all makes sense. Based on the results of that codepen, your tablet in portrait mode is 534px wide, meaning classes targeting @m, @l, @xl don't apply.

Your CV columns are sized via uk-child-width-1-1@s and uk-child-width-1-2@m, so they are acting appropriately.

Your image stamps are not sized with CSS at all as you haven't applied a uk(-child)-width class to those grid items. Those items are therefore being sized automatically based on the width of the content/padding/margin. The content is sized based on the viewbox of the SVG elements, and the padding is coming from your uk-card-body classes.

**Comment by xonixmx** [comment]
Okay, thanks. I fixed some own bugs. Anyway, does it means that @s, @m, etc. params do not works (it's not usefully) for mobile devices, because the real resolution in css pixels is too small and that params  do not take in account PPI of mobile screens (pixel aspect ratio)?

**Comment by xonixmx** [comment]
Thank, Sebastian. I read that and I know about it. Actually, my questions about params @s, @m, etc. in situations with mobile devices (viewport, ppi, pixel ratio, etc). As I can see they're useless for mobile devices, becuase even on the wide screen and landscape orientation on a tablet the with will be too small for that params and I need to write two html code and CSS for different ways -- for PC and mobile devices in this situations. However, I thought before that all of this params works without additional song and dance about it. :-)

**Comment by zzseba78** [comment]
@xonixmx @s and @m and so on are just classes, whey work with media queries. 
"As I can see they're useless for mobile devices", they are created for responsive design, of course they work with mobile devices.
You can set your own breakpoints values easily to feet your needs if you are using less or sass:
https://github.com/uikit/uikit/blob/f2826be6f0f694a5cb95523850c6658a11ce53ab/src/less/components/variables.less#L23-L31
